tuscan/tuscan_figures.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python2
+#
+# Copyright 2016 Kareem Khazem. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Generation of figures from post-processed data."""
+
+from functools import partial
+from logging import basicConfig, debug, info, error, warning, INFO
+from jinja2 import Environment, FileSystemLoader
+from json import load
+from os import listdir, makedirs
+from os.path import basename, join, isdir, splitext
+from subprocess import Popen, PIPE
+from sys import stderr
+
+
+class Counter:
+    def __init__(self, total):
+        self.total = total
+        self.count = 0
+
+    def inc(self):
+        self.count += 1
+        stderr.write("\r%5d/%5d" % (self.count, self.total))
+
+    def finish(self):
+        stderr.write("\n")
+
+
+def render_gnuplot(data, jinja, script_name):
+    template = jinja.get_template("%s.gnu" % script_name)
+    script = template.render(data=data, name=script_name)
+    proc = Popen(["gnuplot"], stdin=PIPE, stdout=PIPE,
```

Comment:
I'm not a big fan of tools that do to many things, like calling `gnuplot` and processing it's output (similarly for invoking `ninja`), I'd rather prefer for Tuscan to just generate the plot and the data file and invoke gnuplot myself.